modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C

```diff
@@ -32,34 +32,43 @@ InteractionIntegralBenchmarkBC::InteractionIntegralBenchmarkBC(const InputParame
     _crack_front_point_index(getParam<unsigned int>("crack_front_point_index")),
     _poissons_ratio(getParam<Real>("poissons_ratio")),
     _youngs_modulus(getParam<Real>("youngs_modulus")),
-    _ki(getParam<Real>("KI")),
-    _kii(getParam<Real>("KII")),
-    _kiii(getParam<Real>("KIII"))
+    _ki_function(&getFunction("KI_function")),
+    _kii_function(&getFunction("KII_function")),
+    _kiii_function(&getFunction("KIII_function"))
```

Comment:
If these are required Functions you should use references (&) instead. That remove the need for pointers and `->` operators below.